src/OrchardCore.Modules/OrchardCore.Taxonomies/TaxonomyContentsAdminListShapes.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+using Newtonsoft.Json.Linq;
+using OrchardCore.ContentManagement;
+using OrchardCore.DisplayManagement;
+using OrchardCore.DisplayManagement.Descriptors;
+using OrchardCore.DisplayManagement.Shapes;
+using OrchardCore.DisplayManagement.Zones;
+using OrchardCore.Entities;
+using OrchardCore.Environment.Shell.Scope;
+using OrchardCore.Modules;
+using OrchardCore.Settings;
+using OrchardCore.Taxonomies.Models;
+using OrchardCore.Taxonomies.Settings;
+using OrchardCore.Taxonomies.ViewModels;
+
+namespace OrchardCore.Taxonomies
+{
+    [Feature("OrchardCore.Taxonomies.ContentsAdminList")]
+    public class TaxonomyContentsAdminListShapes : IShapeTableProvider
+    {
+        private const string LevelPadding = "\xA0\xA0";
+
+        public void Discover(ShapeTableBuilder builder)
+        {
+            builder.Describe("ContentsAdminListHeader")
+                .OnCreated(async context =>
+                {
+                    var shape = (dynamic)context.Shape;
+
+                    var S = ShellScope.Services.GetRequiredService<IStringLocalizer<TaxonomyContentsAdminListShapes>>();
+                    var siteService = ShellScope.Services.GetRequiredService<ISiteService>();
+                    var settings = (await siteService.GetSiteSettingsAsync()).As<TaxonomyContentsAdminListSettings>();
+
+                    var contentManager = ShellScope.Services.GetRequiredService<IContentManager>();
+                    var taxonomies = await contentManager.GetAsync(settings.TaxonomyContentItemIds);
+
+                    var position = 5;
+                    foreach (var taxonomy in taxonomies)
+                    {
+                        var termEntries = new List<FilterTermEntry>();
+                        PopulateTermEntries(termEntries, taxonomy.As<TaxonomyPart>().Terms, 0);
+                        var terms = new List<SelectListItem>
+                            {
+                                new SelectListItem() { Text = S["Clear {0} Filter", taxonomy.DisplayText], Value = ""  },
+                                new SelectListItem() { Text = S["All {0}", taxonomy.DisplayText], Value = "Taxonomy:" + taxonomy.ContentItemId }
```

Comment:
May be just `Show All` because we know which taxonomy we are filtering and it is expecting a plural instead.